pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go

```diff
@@ -305,10 +370,44 @@ func (ci *CWSInstrumentation) injectCWSCommandInstrumentation(exec *corev1.PodEx
 		return false, nil
 	}
 
-	// is the pod instrumentation ready ? (i.e. has the CWS Instrumentation pod admission controller run ?)
-	if !isPodCWSInstrumentationReady(pod.Annotations) {
-		// pod isn't instrumented, do not attempt to override the pod exec command
-		log.Debugf("Ignoring exec request into %s, pod not instrumented yet", common.PodString(pod))
+	var cwsInstrumentationRemotePath string
+
+	switch ci.mode {
+	case InitContainer:
+		cwsInstrumentationRemotePath = filepath.Join(cwsMountPath, "cws-instrumentation")
+		// is the pod instrumentation ready ? (i.e. has the CWS Instrumentation init container been added ?)
+		if !isPodCWSInstrumentationReady(pod.Annotations) {
+			// pod isn't instrumented, do not attempt to override the pod exec command
+			log.Debugf("Ignoring exec request into %s, pod not instrumented yet", common.PodString(pod))
+			return false, nil
+		}
+	case RemoteCopy:
```

Comment:
I'd check also if the container has the security context rootfs readonly

Reply:
I wanted to "let it fail" at the copy step, but you're right, it's cleaner to actually test it 😄 On it !